datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -860,21 +864,41 @@ def refresh_source_node(
     current_revision = source_node.current
 
     # Get the latest columns for the source node's table from the query service
-    columns = query_service_client.get_columns_for_table(
-        current_revision.catalog.name,
-        current_revision.schema_,  # type: ignore
-        current_revision.table,  # type: ignore
-        current_revision.catalog.engines[0]
-        if len(current_revision.catalog.engines) >= 1
-        else None,
-    )
-
-    # Check if any of the columns have changed (only continue with update if they have)
-    column_changes = {col.identifier() for col in current_revision.columns} != {
-        col.identifier() for col in columns
-    }
-    if not column_changes:
-        return source_node
+    new_columns = []
+    try:
+        new_columns = query_service_client.get_columns_for_table(
+            current_revision.catalog.name,
+            current_revision.schema_,  # type: ignore
+            current_revision.table,  # type: ignore
+            current_revision.catalog.engines[0]
+            if len(current_revision.catalog.engines) >= 1
+            else None,
+        )
+    except DJDoesNotExistException:
+        # continue with the update, if the table was not found
+        pass
+
+    refresh_details = {}
+    if new_columns:
+        # check if any of the columns have changed (only continue with update if they have)
+        column_changes = {col.identifier() for col in current_revision.columns} != {
+            col.identifier() for col in new_columns
+        }
+
+        # FIXME: there is a bug with type translation (bigint != long) - fix it. # pylint: disable=fixme
```

Comment:
Ah, so if I understand correctly, this change doesn't handle the bigint vs long translation, but it does manage table existence with the new `missing_table` flag?

Reply:
Yes, that data type translation issue is bigger than I thought and didn't want to block this PR.